satpy/readers/ici_l1b_nc.py

```diff
@@ -0,0 +1,441 @@
+        return observation_azimuth
+
+    @property
+    def cached_observation_zenith(self):
+        """Get observation zenith angles."""
+        _, observation_zenith = self.observation_azimuth_and_zenith
+        return observation_zenith
+
+    @property
+    def cached_solar_azimuth(self):
+        """Get solar azimuth angles."""
+        solar_azimuth, _ = self.solar_azimuth_and_zenith
+        return solar_azimuth
+
+    @property
+    def cached_solar_zenith(self):
+        """Get solar zenith angles."""
+        _, solar_zenith = self.solar_azimuth_and_zenith
+        return solar_zenith
+
+    @property
+    def cached_longitude(self):
+        """Get longitude coordinates."""
+        longitude, _ = self.longitude_and_latitude
+        return longitude
+
+    @property
+    def cached_latitude(self):
+        """Get latitude coordinates."""
+        _, latitude = self.longitude_and_latitude
+        return latitude
+
+    @staticmethod
+    def _calibrate_bt(radiance, cw, a, b):
+        """Perform the calibration to brightness temperature.
+
+        Args:
+            radiance: numpy ndarray containing the radiance values.
+            cw: center wavenumber [cm-1].
+            a: temperature coefficient [-].
+            b: temperature coefficient [K].
+
+        Returns:
+            numpy ndarray: array containing the calibrated brightness
+                temperature values.
+
+        """
+        return b + (a * C2 * cw / np.log(1 + C1 * cw ** 3 / radiance))
+
+    @staticmethod
+    def _perform_geo_interpolation(
+        longitude,
+        latitude,
+        n_samples,
+    ):
+        """
+        Perform the interpolation of geographic coordinates from tie points to pixel points.
+
+        Args:
+            longitude: xarray DataArray containing the longitude dataset to
+                interpolate.
+            latitude: xarray DataArray containing the longitude dataset to
+                interpolate.
+            n_samples: int describing number of samples per scan to interpolate
+                onto.
+
+        Returns:
+            tuple of arrays containing the interpolate values, all the original
+                metadata and the updated dimension names.
+
+        """
+        horns = longitude.n_horns
+        n_scan = longitude.n_scan
+        n_subs = longitude.n_subs
+        lons = np.zeros((n_scan.size, n_samples, horns.size))
+        lats = np.zeros((n_scan.size, n_samples, horns.size))
+        n_subs = np.linspace(0, n_samples - 1, n_subs.size).astype(int)
+        for horn in horns.values:
+            satint = GeoInterpolator(
+                (longitude.values[:, :, horn], latitude.values[:, :, horn]),
+                (n_scan.values, n_subs),
+                (n_scan.values, np.arange(n_samples)),
+            )
+            lons_horn, lats_horn = satint.interpolate()
+            lons[:, :, horn] = lons_horn
+            lats[:, :, horn] = lats_horn
```

Comment:
Is it possible to keep this dask compatible? (I don't know if geotiepoints allows it though)

Reply:
@BengtRydberg did you have on opinion on this?

Reply:
The tests passes if I provide dask arrays to geotiepoints functions, but it seems they run much slower, probably since geotiepoints uses numpy.